pkg/component/etcd/statefulset/values_helper.go

```diff
@@ -119,36 +119,47 @@ func GenerateValues(
 		AutoCompactionRetention: etcd.Spec.Common.AutoCompactionRetention,
 		ConfigMapName:           etcd.GetConfigmapName(),
 		PeerTLSChangedToEnabled: peerTLSChangedToEnabled,
+
+		UseEtcdWrapper: useEtcdWrapper,
 	}
 
 	values.EtcdCommand = getEtcdCommand(values)
 
+	// Use linearizability for readiness probe so that pod is only considered ready
+	// when it has an active connection to the cluster and the cluster maintains a quorum.
+	values.ReadinessProbeCommand = getProbeCommand(values, linearizable)
+
 	values.EtcdBackupCommand = getBackupRestoreCommand(values)
 
 	return values
 }
 
 func getEtcdCommand(val Values) []string {
-	command := []string{"" + "start-etcd"}
-
-	if val.ClientUrlTLS != nil {
-		dataKey := "ca.crt"
-		if val.ClientUrlTLS.TLSCASecretRef.DataKey != nil {
-			dataKey = *val.ClientUrlTLS.TLSCASecretRef.DataKey
+	if val.UseEtcdWrapper {
+		//TODO @aaronfern: remove this feature gate when UseEtcdWraper becomes GA
+		command := []string{"" + "start-etcd"}
+
+		if val.ClientUrlTLS != nil {
+			dataKey := "ca.crt"
+			if val.ClientUrlTLS.TLSCASecretRef.DataKey != nil {
+				dataKey = *val.ClientUrlTLS.TLSCASecretRef.DataKey
+			}
+			command = append(command, "--backup-restore-tls-enabled=true")
+			command = append(command, fmt.Sprintf("--backup-restore-host-port=%s-local:8080", val.Name))
+			command = append(command, fmt.Sprintf("--etcd-server-name=%s-local", val.Name))
+			command = append(command, "--etcd-client-cert-path=/var/etcd/ssl/client/client/tls.crt")
+			command = append(command, "--etcd-client-key-path=/var/etcd/ssl/client/client/tls.key")
+			command = append(command, fmt.Sprintf("--backup-restore-ca-cert-bundle-path=/var/etcd/ssl/client/ca/%s", dataKey))
+		} else {
+			command = append(command, "--backup-restore-tls-enabled=false")
+			command = append(command, fmt.Sprintf("--backup-restore-host-port=%s-local:8080", val.Name))
+			command = append(command, fmt.Sprintf("--etcd-server-name=%s-local", val.Name))
 		}
-		command = append(command, "--backup-restore-tls-enabled=true")
-		command = append(command, fmt.Sprintf("--backup-restore-host-port=%s-local:8080", val.Name))
-		command = append(command, fmt.Sprintf("--etcd-server-name=%s-local", val.Name))
-		command = append(command, "--etcd-client-cert-path=/var/etcd/ssl/client/client/tls.crt")
-		command = append(command, "--etcd-client-key-path=/var/etcd/ssl/client/client/tls.key")
-		command = append(command, fmt.Sprintf("--backup-restore-ca-cert-bundle-path=/var/etcd/ssl/client/ca/%s", dataKey))
-	} else {
-		command = append(command, "--backup-restore-tls-enabled=false")
-		command = append(command, fmt.Sprintf("--backup-restore-host-port=%s-local:8080", val.Name))
-		command = append(command, fmt.Sprintf("--etcd-server-name=%s-local", val.Name))
+
+		return command
 	}
 
-	return command
+	return []string{}
```

Comment:
Can you add a comment here specifying that druid now relies on the default entrypoint from `etcd-custom-image:v3.4.13-bootstrap-12`, which already calls `bootstrap.sh`?